### ISSUE TITLE
Log capturing crashes on run finalization stage

### ISSUE THREAD

**Issue by mihran113** [aimhubio/aim#1790]
## 🐛 Bug

When log capturing is enabled, in some cases it crashes the run finalization process.

```
failed to write
Exception ignored in: <finalize object at 0x101729f60; dead>
Traceback (most recent call last):
  File "/Users/mvanyan/.pyenv/versions/3.6.14/lib/python3.6/weakref.py", line 548, in __call__
    return info.func(*info.args, **(info.kwargs or {}))
  File "/Users/mvanyan/proj/aim/aim/sdk/run.py", line 107, in _close
    self.finalize_system_tracker()
  File "/Users/mvanyan/proj/aim/aim/sdk/run.py", line 93, in finalize_system_tracker
    self._system_resource_tracker.stop()
  File "/Users/mvanyan/proj/aim/aim/ext/resource/tracker.py", line 134, in stop
    self._store_buffered_logs()
  File "/Users/mvanyan/proj/aim/aim/ext/resource/tracker.py", line 216, in _store_buffered_logs
    self._track_func()(log_line, name='logs', step=self._line_counter)
TypeError: 'NoneType' object is not callable
```

### To reproduce

Navigate to an aim repository with more than one existing runs and execute this script.

```
import tqdm
import aim

aim_repo = aim.Repo.default_repo()
aim_runs = list(aim_repo.meta_tree.subtree('chunks').keys())

for run_hash in tqdm(aim_runs):
    aim_run_old = aim_repo.get_run(run_hash)
    aim_run_new = aim.Run(repo='new_aim',)
    experiment_name = None

    for i in tqdm(aim_run_old.iter_metrics_info()):
        metric_name_old = i[0]
        metric_context_old = i[1]

        metric_data_old = aim_run_old.get_metric(metric_name_old, metric_context_old)
```
### Expected behavior

Expected behavior would be no errors at all.

### Environment

- Aim Version: 3.10.1
- Python version: 3.6.14
- pip version:  21.3.1
- OS:  macOS 12.4

**Comment by piojanu** [comment]
I've checked-out and installed your fix #1791. It runs. I'll see tomorrow if the whole sweep finished. Thanks!

**Comment by mihran113** [comment]
> I've checked out and installed your fix #1791. It runs. I'll see tomorrow if the whole sweep will finish. Thanks!
> 
> UPDATE: The sweep finished without errors. However, there is a regression I described here: [#1791 (comment)](https://github.com/aimhubio/aim/pull/1791#issuecomment-1136922509)

Thanks a lot for trying it out and reporting the issues. I've just fixed the regression described in the PR. We'll ship the fix with the next patch release.

**Comment by piojanu** [comment]
It's really great tool and you do great work! Thanks <3 I'll try helping you as much as I can from the user standpoint :)
